app/domain/authentication/authn_k8s/extract_k8s_resource_restrictions.rb

```diff
@@ -0,0 +1,70 @@
+require 'command_class'
+
+module Authentication
+  module AuthnK8s
+
+    ExtractK8sResourceRestrictions = CommandClass.new(
```

Comment:
does this need to be `ExtractK8sResourceRestrictions`? can we name it `ExtractResourceRestrictions` so we have an alignment?

then we have the class `Authentication:: AuthnK8s::ExtractResourceRestrictions` that calls the general class `Authentication::ResourceRestrictions::ResourceRestrictions`.

I would maybe even consider adding a sub-module `ResourceRestrictions ` so we have:
```
Authentication::ResourceRestrictions::ResourceRestrictions
Authentication:: AuthnK8s::ResourceRestrictions::ExtractResourceRestrictions
```

Reply:
I don't like relying on module for class context.
The module is mostly ignored, and it is hard to know which of the classes is the referenced one.
I find it very confusing.
I prefer to distinguish the class by its name, and to treat its module as "family name" - which doesn't say what it does, just where it belong.